File: CS_50/problem_set_1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    
    // ask user to give a height between 0 and 23
    // if user gives bad input, prompts them again
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);
    
    for (int i = 0; i < height; i++)
    {
        // print spaces for left pyramid
        for (int j = 0; j < (height - i - 1); j++)
        {
            printf(" ");
        }
        // print hashes for left pyramid
        for (int k = i; k >= 0; k--)
        {
            printf("#");
        }
        //print gap
        printf("  ");
        // print hashes for right pyramid
        for (int l = i; l >= 0; l--)
        {
            printf("#");
        }
        // print new line
        printf("\n");
    }
}

File: CS_50/problem_set_6/mario.py
import cs50

def main():
    # make sure user gives a positive integer no greater than 23
    while True:
        print("Height: ", end='')
        h = cs50.get_int()
        if (h >= 0 and h <= 23):
            break
    
    # draws the pramid
    draw_pyramid(h)

# draws the pyramid based on the h (height) given
def draw_pyramid(h):
    # go through each step of pyramid
    for i in range(h):
        # print spaces for left pyramid
        for j in range(h - i - 1):
            print(' ', end='')
        
        # print hashes for left pyramid 
        for k in range(i + 1):
            print('#', end='')
        
        # gap between left and right pyramid
        print(' ', end='')
        
        # print hashes for right pyramid
        for l in range(i+1):
            print('#', end='')
        
        # this line is done, move onto next line
        print()

if __name__ == "__main__":
    main()